AEPCore/Tests/MobileCoreInitializerTests.swift

```diff
@@ -0,0 +1,416 @@
+/*
+ Copyright 2025 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+
+import XCTest
+
+@testable import AEPCore
+@testable import AEPCoreMocks
+@testable import AEPServices
+
+class MobileCoreInitializerTests: XCTestCase {
+    override func setUp() {
+        NamedCollectionDataStore.clear()
+        MobileCore.resetSDK()
+        MockExtension.reset()
```

Comment:
Can we create a mocked event hub and test the API calls directly, rather than registering a mock extension and adding listeners to it? In general, it would be simpler if we could minimize the number of `EventHub.shared` calls in individual tests?